src/Test/Polysemy/Mock/TH.hs

```diff
@@ -110,7 +110,11 @@ mkInitialReturns c =
 
 mkMockMatch :: Type -> ConLiftInfo -> Match
 mkMockMatch t c =
+#if MIN_VERSION_template_haskell(2,18,0)
+  let pat = ConP (cliConName c) (map snd (cliFunArgs c)) (map (VarP . fst) (cliFunArgs c))
```

Comment:
```suggestion
  let pat = ConP (cliConName c) [] (map (VarP . fst) (cliFunArgs c))
```

I think this expression doesn't need type application. So, maybe its better to leave this list of type applications as empty list.

Reply:
Okay, I've made the suggested changes.